indexer-app/src/cmr/indexer/data/concepts/variable.clj

```diff
@@ -64,18 +64,17 @@
                                           "revision-id" (:associated-revision-id va)}))
                                       variable-associations))))})))
 
-(defn- variable-association->variable-concept
-  "Returns the variable concept and variable association for the given variable association."
-  [context variable-association]
-  (let [{:keys [variable-concept-id]} variable-association
-        variable-concept (mdb/find-latest-concept
-                          context
-                          {:concept-id variable-concept-id}
-                          :variable)]
-    (when-not (:deleted variable-concept)
-      ;; associate variable association into variable concept, so we can use it to generate
-      ;; elastic document for nested variables field
-      (assoc variable-concept :variable-association variable-association))))
+(defn- variable-associations->variable-concepts
+  "Returns the variable concepts for the given variable associations."
+  [context variable-associations]
+  (let [variable-concept-ids (map :variable-concept-id variable-associations)
+        id->association (zipmap variable-concept-ids variable-associations)
+        add-association-fn (fn [concept]
+                             (assoc concept :variable-association (get id->association (:concept-id concept))))
+        variable-concepts (mdb/get-latest-concepts context variable-concept-ids true)]
```

Comment:
I added `true` parameter to allow missing concepts. This is just defensive coding against unlikely scenario of corrupted database. In normal cases, every variable association should be linked to an existing variable concept.

Reply:
I'm definitely against defensive programming -- however! We've been hit pretty hard in the past with data corruption issues, so I'm a big +1 on this :-)